viewing_party/main.py

```diff
@@ -0,0 +1,240 @@
+    if title and genre and rating:
+        movie_dict = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data = {
+        "watched": []
+    }
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {
+        "watchlist": []
+    }
+    user_data["watchlist"].append(movie)
+    return user_data
+
+'''
+If title is in a movie in the user's watchlist, remove that movie
+from watchlist and add that to watched.
+'''
+def watch_movie(user_data, title):
+
+    user_data_copy = user_data.copy()
+
+    for movie in user_data_copy["watchlist"]: # initial length: 2
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+            # break #Do I always need a break in a for-else loop? Or return will stop?
+                # else:
+                #     return user_data
+            # return use_data
+        # return use_data
+    else:
+        return user_data
+
+# Wave 2
+def get_watched_avg_rating(user_data):
+
+    # for v in user_data.values():
+    #     if v == []:
+    #         return 0.0
+    #     else:
+    #         for movie in v:
+    #             total_rating += movie["rating"]
+    #         avg = total_rating / len(v)
+    #         return avg
+
+    if user_data["watched"] == []:
+        return 0.0
+
+    total_rating = 0
+    for movie in user_data["watched"]:
+            total_rating += movie["rating"]
+            avg = total_rating / len(user_data["watched"])
+    return avg
+
+def get_most_watched_genre(user_data):
+    genre_map = {}
+    for movie in user_data["watched"]:
+        x = movie["genre"]
+        if x not in genre_map:
+            genre_map[x] = 1
+        else:
+            genre_map[x] += 1
+    
+    most_watched_genre = (max(genre_map, key=genre_map.get)) if genre_map else None
+
+    return most_watched_genre 
+
+    # current_max = 0
+    # for k, v in genre_map.items():
+    #     if v > current_max:
+    #         current_max = v
+    
+    # for k, v in genre_map.items():
+    #     if v == current_max:
+    #         return k
+
+# Wave 3
+'''
+Determine which movies the user has watched, 
+but none of their friends have watched.
+'''
+def get_unique_watched(user_data):
+
+    mine_watched = {}
+    for movie in user_data["watched"]:
+        mine_watched[movie["title"]] = True
+    
+    friends_watched = {}
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched[movie["title"]] = True
+
+    diff = []
+    for title in mine_watched.keys():
+        if title not in friends_watched:
+            diff.append({"title": title})
+    return diff
```

Comment:
This will pass the tests because the tests for this wave use a simplified version of the movie dictionary.  However, the movie dictionary can have a 'title', 'genre' and 'rating'.  How could you use the `diff` list to generate a list of the original dictionaries?